mne/simulation/raw.py

```diff
@@ -0,0 +1,511 @@
+                                 _get_stim_channel(None, raw.info))[0]
+        raw_data[event_ch, :] = 0.
+        logger.info('Event information will be stored in "%s"'
+                    % raw.ch_names[event_ch])
+    else:
+        logger.info('Event information will not be stored')
+    used = np.zeros(len(times), bool)
+    stc_indices = np.arange(len(times)) % len(stc.times)
+    raw_data[meeg_picks, :] = 0.
+    hpi_mag = 70e-9
+    last_fwd = last_fwd_chpi = last_fwd_blink = last_fwd_ecg = src_sel = None
+    zf = None  # final filter conditions for the noise
+    # don't process these any more if no MEG present
+    ecg = ecg and len(meg_picks) > 0
+    chpi = chpi and len(meg_picks) > 0
+    if chpi:
+        chpi_rrs = np.array([d['r'] for d in info['dig']
+                            if d['kind'] == FIFF.FIFFV_POINT_HPI])
+        chpi_nns = chpi_rrs / np.sqrt(np.sum(chpi_rrs * chpi_rrs,
+                                             axis=1))[:, np.newaxis]
+    for fi, (fwd, fwd_blink, fwd_ecg, fwd_chpi) in \
+        enumerate(_iter_forward_solutions(
+            fwd_info, trans, src, bem, exg_bem, dev_head_ts, mindist,
+            chpi_rrs, blink_rr, ecg_rr, n_jobs)):
+        # must be fixed orientation
+        fwd = convert_forward_solution(fwd, surf_ori=True,
+                                       force_fixed=True, verbose=False)
+        # just use one arbitrary direction
+        if blink:
+            fwd_blink = fwd_blink['sol']['data'][:, ::3]
+        if ecg:
+            fwd_ecg = fwd_ecg['sol']['data'][:, ::3]
+
+        # align cHPI magnetic dipoles in approx. radial direction
+        if chpi:
+            for ii in range(len(chpi_rrs)):
+                fwd_chpi[:, ii] = np.dot(fwd_chpi[:, 3 * ii:3 * (ii + 1)],
+                                         chpi_nns[ii])
+            fwd_chpi = fwd_chpi[:, :len(chpi_rrs)].copy()
+
+        if src_sel is None:
+            src_sel = _stc_src_sel(fwd['src'], stc)
+            verts = stc.vertices
+            verts = [verts] if isinstance(stc, VolSourceEstimate) else verts
+            diff_ = sum([len(v) for v in verts]) - len(src_sel)
+            if diff_ != 0:
+                warnings.warn('%s STC vertices omitted due to fwd calculation'
+                              % (diff_,))
+        if last_fwd is None:
+            last_fwd, last_fwd_blink, last_fwd_ecg, last_fwd_chpi = \
+                fwd, fwd_blink, fwd_ecg, fwd_chpi
+            continue
+
+        # set up interpolation
+        n_pts = offsets[fi] - offsets[fi - 1]
+        if interp == 'zero':
+            interps = None
+        else:
+            if interp == 'linear':
+                interps = np.linspace(1, 0, n_pts, endpoint=False)
+            else:  # interp == 'cos2':
+                interps = np.cos(0.5 * np.pi * np.arange(n_pts)) ** 2
+            interps = np.array([interps, 1 - interps])
+
+        assert not used[offsets[fi - 1]:offsets[fi]].any()
```

Comment:
no, this ensures that the points that are about to be written/saved/generated have not been done yet, the `.all()` comes later